### DISCUSSION TITLE
Welcome to hekili Discussions!

### DISCUSSION THREAD

**Discussion by Hekili** [Hekili/hekili#652] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Crepette** [comment]
Hello, I have been using Hekili for a little while, and I was wondering if you have any video on how to use this addon to its fullest.  As an example is it better to have the display mode as reactive or to change it each time for either single target or AOE Thanks

**Reply by Hekili** [reply]
Hi, I don't presently have a video but there are some videos that other folks have created, on YouTube.

For display mode, "Automatic" is generally recommended and works very well for melee specs.  For ranged, sometimes it's better to use Dual so that you have single-target recommendations and AOE recommendations at the same time.  I use Dual when I'm playing Shadow Priest in M+, for example, because I can use the ST recommendations while focusing down a target or the AOE recommendations when trying to melt a whole pack.

Using the display mode toggle, you can swap between modes as you see fit, as well.

**Comment by harsjo** [comment]
Hello!

Great addon, I very much appreciate your work!

One of the best things about it, in my opinion, is the configurability. I've spent many hours tinkering with my action lists over the years.

In shadowlands I started playing a mistweaver monk and activated the mistweaver action pack. Since it is not yet supported I find myself missing a few things and tried diving into the code to try and add what I wanted. I am by no means a programmer and only know the basics pretty much, so it did not go very well. (That's another great thing about this addon, I get to practice some programming in a very simple and fun way!)

I more specifically want the buff "Ancient teachings of the monastery" from the legendary with the same name to exist in the addon. So I tried adding 
```
         ancient_teachings_of_the_monastery = {
            id = 202899,
            duration = 15,
            max_stack = 1,
},
```
inside MonkMistweaver.lua with the other aura registrations and then added `applyBuff( "ancient_teachings_of_the_monastery" )` in the handler of Tiger Palm (for testing).

The addon sort of seems to get what I want but not quite. The next suggested ability seems to think the buff gets added when I cast Tiger Palm but after I do cast it the suggestion resets, as if the buff doesn't actually get applied.


I'm probably doing a lot of things wrong (for example the aura-id should probably not just be a random number?) but I was wondering if you could give me a few pointers? Is my line of thinking completely off or am I close to figuring out how to add buffs and abilities to the source code?

Many thanks again!

**Reply by Hekili** [reply]
202899 is not the spell ID for Ancient Teachings of the Monastery.

https://www.wowhead.com/spell=347553/ancient-teachings-of-the-monastery looks like the correct buff, ID 347553.

Mechanically speaking, you'd want to `applyBuff( "ancient_teachings_of_the_monastery" )` in Essence Font's handler.  If the buff is consumed on your next cast of Tiger Palm, etc., then `removeBuff( "ancient_teachings_of_the_monastery" )` in the handlers for those abilities.  (I genuinely don't know if the buff just lasts for 15 seconds or if it gets consumed by the next cast, sorry.)

**Reply by harsjo** [reply]
Sweet, the correct ID did the trick. Thank you!

I can't get the buff part (reducing the cost and cast time of your next Enveloping Mist by 33%, stacking) of https://www.wowhead.com/spell=325197/invoke-chi-ji-the-red-crane#summons to work either, probably because of wrong ID (I tried 343821) but the correct aura doesn't seem to exist in wowdb at all, is that possible? If so, is it possible to add an aura to the addon without a spell ID?

Also: what's the syntax for applying a buff only when pet is active? Neither `if pet.alive then` nor `if pet.exists then` seem to work (with chiji at least).

**Reply by Hekili** [reply]
Honestly, that's kinda getting to the point where I'm writing the unsupported module, which defeats the "unsupported" part of its description.

There are similar mechanics elsewhere, but pet/guardians are tricky.  You might look at how BrM/WW have set up Ox/Xuen; there are differences in how pets need to be detected between pets vs. guardians, and whether I can use NPC ID or texture to try to track them.  But you'd really be on your own to solve that unless/until I build those healer spec modules myself.

**Reply by harsjo** [reply]
Alright, I understand. I'll keep tinkering. Thanks again and hope you keep up the great work for a long time to come. :)

**Comment by LucianDK** [comment]
Hi. Ive just started to use Hekili. Is there options like hiding it out of combat or active when targetting a hostile? No reason to have it up in like a sanctuary zone.
And is it possible to drag the displays instead of relying on XY coords?

**Reply by Hekili** [reply]
If you move the options window aside, you can click and drag displays where you want.

Under Displays > Primary > Visibility, you can check Advanced and specify opacity in different scenarios.  0 = hidden.

**Reply by LucianDK** [reply]
Thank you, exactly what I was looking for!

**Comment by LucianDK** [comment]
Hello again. So far Ive had great use of your addon. But one thing that needs work is the target swapping indicator. The tiny indicator is really difficult to notice when in the heat of fighting.
I am an Elemental Shaman, and its Flame Shock for me that needs it.

Perhaps giving the primary icon some sort of colored pulsing aura to make it noticeable that you should swap NOW! Or maybe shaking the primary icon.

**Reply by Hekili** [reply]
Target swapping is already controversial enough, really.  I have ideas for when I make multitarget recommendations more versatile (probably color coding targets and using a colorful border or colorful marker).  Otherwise, gonna leave it as-is for the time being.

**Reply by LucianDK** [reply]
Ah thank you. Another question.

I cant seem to add my primal fire elemental's meteor spell to the queue displayed in the addon, which is part of the roation.

**Reply by Hekili** [reply]
It's supported and should be listed if you make a custom priority.  Manual control of Meteor isn't in the sim priority at this time.

**Comment by JamesLJ1** [comment]
Hi Hekili, I am having a lot of success with your addon after being recommended it by a guildie. I have a quick question re: Outlaw rogue. The addon rarely seems to suggest Between the Eyes while doing open world quests (It has quite a high priority on the icyveins guide). I have a look into the priority set up on the addon and I believe it is basically telling me that the advantage of Between the Eyes crit bonus isn't worth it as the mob will die quick enough that Dispatch is more beneficial. Is this right? It makes sense to me but just wanted to clarify in case I am missing something. I guess I am asking if Between the Eyes is a wasted finisher in quick overworld fights? Many thanks!

**Reply by Hekili** [reply]
More or less, yes.  If you're not going to get the benefit from the debuff, you might as well just Dispatch.

**Reply by JamesLJ1** [reply]
Thank you for the reply, makes a lot of sense. Just one additional question in regards to Blade Rush, it often seems to get pushed back in the queue and curious is this due to energy levels? I couldn't figure out on the priority list how it calculates when Blade Rush is best used!
